src/Illuminate/Database/MigrationServiceProvider.php

```diff
@@ -67,15 +93,133 @@ protected function registerCreator()
         });
     }
 
+    /**
+     * Register the given commands.
+     *
+     * @param  array  $commands
+     * @return void
+     */
+    protected function registerCommands(array $commands)
+    {
+        foreach (array_keys($commands) as $command) {
```

Comment:
You can move array_keys out of foreach on a variable to avoid executing the function in each iteration

Reply:
foreach excecutes once given expression, so array_keys is called only once

Reply:
sorry, I understood that for each iteration the function is executed if it is used as a value to iterate.